src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Logging/MessageOrigin.cs

```diff
@@ -3,45 +3,53 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection.Metadata;
 using System.Text;
 using Internal.IL;
 using Internal.TypeSystem;
 
+#nullable enable
+
 namespace ILCompiler.Logging
 {
-    public struct MessageOrigin
+    public struct MessageOrigin :
 #if false
-        : IComparable<MessageOrigin>, IEquatable<MessageOrigin>
+        IComparable<MessageOrigin>,
```

Comment:
It was there before, so I kept it for now. I agree that we may eventually remove it, depends on how exact we want to be with the linker. Linker sorts all warnings before printing them out. I think it would make sense to do the same for ILC - even more so, since ILC is multi-threaded and thus more likely to produce warnings in random order. We should catch this as we port more tests over.

Reply:
The `#if false` is equivalent to commenting out code. I used `#if false` for larger blocks of code here. I tend to keep commented out code for stuff that consciously wasn't ported over for now.

We track determinism as "the output files should be the same". We haven't really bothered with ordering the warnings. Sometimes it's nice to get warnings from the tool even before the tool finishes. One can just "Ctrl-C" the tool and go fix the warning if it's relevant.